common/src/main/java/com/wynntils/models/gear/GearInfoRegistry.java

```
@@ -379,39 +379,34 @@ private Optional<String> parseQuest(JsonObject json) {
+            Optional<GearMajorId> majorIds = parseMajorIds(json);
+            List<Pair<DamageType, RangedValue>> damages = parseDamages(baseStats);
+            List<Pair<Element, Integer>> defences = parseDefences(baseStats);
 
             return new FixedStats(healthBuff, skillBonuses, attackSpeed, majorIds, damages, defences);
         }
 
-        private List<GearMajorId> parseMajorIds(JsonObject json) {
-            JsonElement majorIdsJson = json.get("majorIds");
-            if (majorIdsJson == null || majorIdsJson.isJsonNull()) return List.of();
+        private Optional<GearMajorId> parseMajorIds(JsonObject json) {
+            JsonObject majorIdsJson = JsonUtils.getNullableJsonObject(json, "majorIds");
+            if (majorIdsJson == null || majorIdsJson.isJsonNull() || majorIdsJson.isEmpty()) return Optional.empty();
 
-            return majorIdsJson.getAsJsonArray().asList().stream()
-                    .map(majorIdName -> getMajorIdFromString(majorIdName.getAsString()))
-                    .filter(Objects::nonNull)
-                    .toList();
-        }
-
-        private GearMajorId getMajorIdFromString(String majorIdString) {
-            return this.allMajorIds.stream()
-                    .filter(mId -> mId.id().equals(majorIdString))
-                    .findFirst()
-                    .orElse(null);
+            return Optional.of(new GearMajorId(
+                    majorIdsJson.get("name").getAsString(),
+                    StyledText.fromString(majorIdsJson.get("description").getAsString())));
         }
 
         private List<Pair<Skill, Integer>> parseSkillBonuses(JsonObject json) {
             List<Pair<Skill, Integer>> list = new ArrayList<>();
             for (Skill skill : Skill.values()) {
-                String skillBonusApiName = skill.getApiName() + "Points";
+                String skillBonusApiName = "raw" + com.wynntils.utils.StringUtils.capitalizeFirst(skill.getApiName());
```

Okaaay... stat key names have been a constant source of error. Have you verified that this works properly on the new data? Or rather, more generally, have you checked that you do not encounter stat key names that you cannot properly parse? I remember that we had a problem before of just silently ignoring stat keys that we parsed incorrectly, since we did not actually parse every single field in the json object, instead we construct keys we want to look at to pick up.

I remember vaguely that I re-created the json and compared it to the one I read in. I don't remember if I kept that as part of the code, or if it was just debug code that I threw away. Anyway, if we don't have that right now, I recommend you re-instate it.

---

No, the new naming scheme is quite clear now, and it does not seem to have edge-cases. I have not checked every misc stat by hand though, however I do plan to do that after I also ported ingredients. Then I would most likely write some kind of tool to check stat names, instead of scrolling a 100 pages of trade market :)